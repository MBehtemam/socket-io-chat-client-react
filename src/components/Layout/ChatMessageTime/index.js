import React, { Component } from "react";
import styled from "styled-components";

const Time = styled.time``;

export default class ChatMessageTime extends Component {
  render() {
    const { time, twelveMode } = this.props;
    const d = new Date(time);
    const t = twelveMode
      ? `${d.getHours() - 12}:${d.getMinutes()}`
      : `${d.getHours()}:${d.getMinutes()}`;
    return <Time>{t}</Time>;
  }
}
