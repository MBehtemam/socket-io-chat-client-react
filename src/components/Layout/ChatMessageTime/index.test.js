import React from "react";
import styled from "styled-components";
import renderer from "react-test-renderer";
import "jest-styled-components";
import ChatMessageTime from "./";

describe("Test suits for ChatMessageTime", () => {
  test("should match with snapshot", () => {
    const tree = renderer.create(<ChatMessageTime />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
