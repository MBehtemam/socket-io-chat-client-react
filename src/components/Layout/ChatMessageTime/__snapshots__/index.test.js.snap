// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for ChatMessageTime should match with snapshot 1`] = `
<time
  className=""
>
  NaN:NaN
</time>
`;
